### PR TITLE
Authorization Middleware

### PR DESCRIPTION
**Pull request by bbengfort**
Adds middleware that checks the Bearer JWT token from the Authorization
header before permitting the request to continue.

TODO:

- [x] Add middleware to all endpoints but the `authenticate` and `reauthenticate` endpoints 
- [x] Determine if authorization should happen before double cookie verification
- [x] Add claims to the context so that the request has access to the user email address for auditing
- [x] Create unit tests for the middleware 
- [ ] Verify the endpoints generally work (we'll need to work on integration tests in the next epic)

### CODE REVIEW

File: pkg/gds/admin.go [hunk 1]
```diff
@@ -132,6 +132,7 @@ func (s *Admin) setupRoutes() (err error) {
 	// Application Middleware
 	s.router.Use(ginzerolog.Logger("gin"))
 	s.router.Use(gin.Recovery())
+	s.router.Use(s.Available())
```
Comment (pdeziel) [inline pkg/gds/admin.go:135]: Any particular reason why this was moved? Does the order of these statements matter or is it mainly cosmetic?

**Review by pdeziel — APPROVED** [review]
Looks good to me, are there any other security measures we are planning to add to the authorization middleware?

File: pkg/gds/admin.go (hunk 1 above)
Comment (bbengfort) [reply pkg/gds/admin.go:135]: I think it actually does matter; the order of the middleware application determines the order middleware is applied.

**Comment by bbengfort** [comment]
Thanks @pdeziel! We will probably continue to refine our security mechanisms but for now this should be enough to get us to an MVP!

**Pull request merged**